app-android-xamarin/Moodis/Moodis/Feature/Menu/MenuActivity.cs

```diff
@@ -114,18 +113,28 @@ public void LogoutWindowShow()
             builder.Create().Show();
             builder.Dispose();
         }
-        private void InitButtons()
+
+        public bool OnNavigationItemSelected(IMenuItem item)
         {
-            var btnHistory = FindViewById(Resource.Id.buttonHistory);
-            var btnPlayMusic = FindViewById(Resource.Id.playMusic);
-            var btnStopMusic = FindViewById(Resource.Id.StopMusic);
-            var btnGroups = FindViewById(Resource.Id.groups);
+            int id = item.ItemId;
 
-            btnHistory.Click += (sender, e) =>
+            if (id == Resource.Id.nav_camera)
+            {
+                StartActivity(new Intent(this, typeof(CameraActivity)));
```

Comment:
You are doing the same mistake here regarding activity back stack. Just test it by yourself. Continously open menu activity, then click on camera in nav drawer, take picture, go to menu activity again. Do this repeatedly for a few times, then click back button repeatedly. You'll notice that you have not one, but many menu and camera activities opened.